modules/ppcp-settings/src/Endpoint/CommonRestEndpoint.php

```diff
@@ -103,13 +117,23 @@ public function register_routes() {
 	 *
 	 * @return WP_REST_Response The common settings.
 	 */
-	public function get_details() : WP_REST_Response {
+	public function get_details(): WP_REST_Response {
 		$js_data = $this->sanitize_for_javascript(
 			$this->settings->to_array(),
 			$this->field_map
 		);
 
-		return $this->return_success( $js_data );
+		$js_flags = $this->sanitize_for_javascript(
+			$this->settings->get_flags(),
+			$this->flag_map
+		);
+
+		return $this->return_success(
+			$js_data,
+			array(
+				'flags' => $js_flags,
```

Comment:
Possibly, the attribute name `flags` could be a bit misleading. Let's change this to something like `wooSettings` (or `woo`, etc) to make it clearer what data the object contains.